### PR TITLE
for real this time, added tps sync

### PR DESCRIPTION
**Pull request by HausemasterIssue**
accept it this time i tested it and it should work

### CODE REVIEW

**Comment by Sxmurai** [comment]
did you make this yourself? or what client did you take it from, if you took it from somewhere id like to know so i can credit them in the README

**Comment by HausemasterIssue** [comment]
> 
> 
> did you make this yourself? or what client did you take it from, if you took it from somewhere id like to know so i can credit them in the README

no i pasted the getTPS and based the sync TPS off of cookie client :P

**Comment by HausemasterIssue** [comment]
https://github.com/bebeli555/CookieClient

File: src/main/java/me/sxmurai/inferno/features/modules/player/Timer.java [hunk 1]
```diff
@@ -3,11 +3,31 @@
...
 import me.sxmurai.inferno.features.settings.Setting;
 import me.sxmurai.inferno.managers.modules.Module;
+import net.minecraft.util.math.MathHelper;
 import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
 
 @Module.Define(name = "Timer", description = "Makes the game go zoom zoom", category = Module.Category.PLAYER)
 public class Timer extends Module {
-    public final Setting<Float> speed = this.register(new Setting<>("Speed", 2.0f, 0.1f, 20.0f));
+    public final Setting<Float> speed = this.register(new Setting<>("ClientTPS", 2.0f, 0.1f, 20.0f));
+    public final Setting<Boolean> sync = this.register(new Setting<>("TPS Sync", true));
```
Comment (Sxmurai) [inline src/main/java/me/sxmurai/inferno/features/modules/player/Timer.java:12]: make it TPSSync (consistency with the other settings) please

Comment (Sxmurai) [reply src/main/java/me/sxmurai/inferno/features/modules/player/Timer.java:12]: the name of the setting, not the field name

File: src/main/java/me/sxmurai/inferno/features/modules/player/Timer.java [hunk 2]
```diff
@@ -3,11 +3,31 @@
...
+    
+    private static float[] ticks = new float[20];
```
Comment (Sxmurai) [inline src/main/java/me/sxmurai/inferno/features/modules/player/Timer.java:14]: lines 14-29 should become a manager, i'll add the Server manager in my next commit and i'll tell you what to do from there

File: src/main/java/me/sxmurai/inferno/features/modules/player/Timer.java [hunk 3]
```diff
@@ -25,6 +45,12 @@ protected void onDeactivated() {
 
     @SubscribeEvent
     public void onUpdate(UpdateEvent event) {
-        mc.timer.tickLength = 50.0f / speed.getValue();
+    	if(sync.getValue() == true) {
```
Comment (Sxmurai) [inline src/main/java/me/sxmurai/inferno/features/modules/player/Timer.java:48]: the `== true` isn't needed as `sync` is a boolean, so it infers by the value of the boolean already

**Review by Sxmurai — CHANGES_REQUESTED** [review]
please read through this

File: src/main/java/me/sxmurai/inferno/features/modules/player/Timer.java (hunk 3 above)
Comment (Sxmurai) [reply src/main/java/me/sxmurai/inferno/features/modules/player/Timer.java:48]: also add a space between the `if` and the (, again consistency

File: src/main/java/me/sxmurai/inferno/features/modules/player/Timer.java [hunk 4]
```diff
@@ -45,7 +45,7 @@ protected void onDeactivated() {
 
     @SubscribeEvent
     public void onUpdate(UpdateEvent event) {
-    	if(sync.getValue() = true) {
+    	if(sync.getValue()) {
```
Comment (Sxmurai) [inline src/main/java/me/sxmurai/inferno/features/modules/player/Timer.java:48]: add a space between the if and the ( please

**Comment by HausemasterIssue** [comment]
i did the crap now

File: src/main/java/me/sxmurai/inferno/features/modules/player/Timer.java [hunk 5]
```diff
@@ -3,11 +3,31 @@
...
 
 @Module.Define(name = "Timer", description = "Makes the game go zoom zoom", category = Module.Category.PLAYER)
 public class Timer extends Module {
-    public final Setting<Float> speed = this.register(new Setting<>("Speed", 2.0f, 0.1f, 20.0f));
+    public final Setting<Float> speed = this.register(new Setting<>("ClientTPS", 2.0f, 0.1f, 20.0f));
+    public final Setting<Boolean> sync = this.register(new Setting<>("TPSSync", true));
+    
+    private static float[] ticks = new float[20];
+    
+    public static double getTps() {
```
Comment (Sxmurai) [inline src/main/java/me/sxmurai/inferno/features/modules/player/Timer.java:16]: last thing, remove this entire method, replace the getTps() with `Inferno.serverManager.getTps()` and then you can remove the float cast, as the `getTps` method returns a float

**Comment by HausemasterIssue** [comment]
should i change my stuff to work with the server manager?

File: src/main/java/me/sxmurai/inferno/features/modules/player/Timer.java [hunk 6]
```diff
@@ -3,11 +3,31 @@
...
+import net.minecraft.util.math.MathHelper;
 import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
 
 @Module.Define(name = "Timer", description = "Makes the game go zoom zoom", category = Module.Category.PLAYER)
 public class Timer extends Module {
-    public final Setting<Float> speed = this.register(new Setting<>("Speed", 2.0f, 0.1f, 20.0f));
+    public final Setting<Float> speed = this.register(new Setting<>("ClientTPS", 2.0f, 0.1f, 20.0f));
+    public final Setting<Boolean> sync = this.register(new Setting<>("TPSSync", true));
+    
+    private static float[] ticks = new float[20];
```
Comment (Sxmurai) [inline src/main/java/me/sxmurai/inferno/features/modules/player/Timer.java:14]: remove this too please

Comment (HausemasterIssue) [reply src/main/java/me/sxmurai/inferno/features/modules/player/Timer.java:14]: ok

**Comment by HausemasterIssue** [comment]
it is done

File: src/main/java/me/sxmurai/inferno/features/modules/player/Timer.java [hunk 7]
```diff
@@ -46,7 +28,7 @@ protected void onDeactivated() {
     @SubscribeEvent
     public void onUpdate(UpdateEvent event) {
     	if (sync.getValue()) {
-            mc.timer.tickLength = (float) (50.0f / getTps());
+            mc.timer.tickLength = (50.0f / Inferno.serverManager.getTps());
```
Comment (Sxmurai) [inline src/main/java/me/sxmurai/inferno/features/modules/player/Timer.java:31]: i forgot, instead of using `50.0f` do `1000.0f` please lol, thats the last thing and i'll merge your pr

Comment (HausemasterIssue) [reply src/main/java/me/sxmurai/inferno/features/modules/player/Timer.java]: done

**Review by HausemasterIssue — COMMENTED** [review]
ok

File: src/main/java/me/sxmurai/inferno/features/modules/player/Timer.java [hunk 8]
```diff
@@ -25,6 +27,12 @@ protected void onDeactivated() {
 
     @SubscribeEvent
     public void onUpdate(UpdateEvent event) {
-        mc.timer.tickLength = 50.0f / speed.getValue();
+    	if (sync.getValue()) {
+            mc.timer.tickLength = (1000.0f / Inferno.serverManager.getTps());
+    	} else {
+            mc.timer.tickLength = 1000.0f / speed.getValue();
```
Comment (Sxmurai) [inline src/main/java/me/sxmurai/inferno/features/modules/player/Timer.java:33]: this should be 50.0f, i only meant the one where you do the TPS, sorry and i promise that should be it
